File: AN_2/SEM_1/MAP/lab2/src/container/AbstractContainer.java
package container;

import model.Task;

public abstract class AbstractContainer implements Container
{
    protected Task[] taskuri;
    protected int size;

    public void add(Task task)
    {
        if(taskuri.length == size)
        {
            Task[] t= new Task[taskuri.length*2];
            System.arraycopy(taskuri, 0, t, 0, taskuri.length);
            taskuri = t;
        }
        taskuri[size++] = task;
        size++;
    }

    public int size()
    {
        return size;
    }

    public boolean isEmpty()
    {
        return size == 0;
    }
}

File: AN_2/SEM_1/MAP/lab9/lab9/Container/AbstractContainer.cs
using Model;

namespace Container
{
    public abstract class AbstractContainer : IContainer
    {
        protected Tasks[] taskuri;
        protected int size;

        public void Add(Tasks task)
        {
            if (taskuri.Length == size)
            {
                Tasks[] t = new Tasks[taskuri.Length * 2];
                System.Array.Copy(taskuri, 0, t, 0, taskuri.Length);
                taskuri = t;
            }
            taskuri[size++] = task;
        }

        public int Size()
        {
            return size;
        }

        public bool IsEmpty()
        {
            return size == 0;
        }

        public abstract Tasks Remove();
    }
}
